From d81d82dbea310f8491ec7a6e2cbcb78516c52b8d Mon Sep 17 00:00:00 2001
From: Gleb Mazovetskiy <glex.spb@gmail.com>
Date: Fri, 4 Jul 2025 18:44:27 +0100
Subject: [PATCH 1/9] Reduce `cmake_minimum_required` to 3.22

Ubuntu 22.04 CMake is 3.22.
Debian stable CMake is 3.25.
Debian oldstable (EOL in 2026) has CMake 3.25 in bullseye-backports.
---
 CMakeLists.txt               | 2 +-
 documentation/CMakeLists.txt | 2 +-
 single/CMakeLists.txt        | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index b54f71a3..507d9be5 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -22,7 +22,7 @@
 
 # # # # sol2
 # # # Required minimum version statement
-cmake_minimum_required(VERSION 3.26.0)
+cmake_minimum_required(VERSION 3.22)
 # # # Project Include - file that is included after project declaration is finished
 set(CMAKE_PROJECT_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Includes/Project.cmake")
 
diff --git documentation/CMakeLists.txt documentation/CMakeLists.txt
index 1c7b6758..ba780c0f 100644
--- documentation/CMakeLists.txt
+++ documentation/CMakeLists.txt
@@ -22,7 +22,7 @@
 
 # # # # sol2, documentation generation
 # # # Required minimum version statement
-cmake_minimum_required(VERSION 3.26.0)
+cmake_minimum_required(VERSION 3.22)
 	
 find_package(Doxygen REQUIRED)
 find_package(Python3 REQUIRED)
diff --git single/CMakeLists.txt single/CMakeLists.txt
index 2d55fe32..a16926de 100644
--- single/CMakeLists.txt
+++ single/CMakeLists.txt
@@ -22,7 +22,7 @@
 
 # # # # sol3, single
 # # # Required minimum version statement
-cmake_minimum_required(VERSION 3.26.0)
+cmake_minimum_required(VERSION 3.22)
 
 find_package(Python3 REQUIRED)
 
-- 
2.48.1

